frontend/public/src/components/CourseProductDetailEnroll_test.js

```diff
@@ -532,12 +534,95 @@ describe("CourseProductDetailEnrollShallowRender", () => {
       if (multiples) {
         assert.isTrue(selectorControl.exists())
         const selectorControlItems = selectorControl.find("option")
-        assert.isTrue(selectorControlItems.length === 2)
+        assert.isTrue(selectorControlItems.length === 3)
+        assert.isTrue(selectorControlItems.at(0).text() === "Please Select")
+        assert.isTrue(
+          upgradeForm
+            .find("button.btn-upgrade")
+            .at(0)
+            .prop("disabled")
+        )
+        assert.isTrue(
+          inner
+            .find("button.enroll-now-free")
+            .at(0)
+            .prop("disabled")
+        )
+        modal
+          .find("select.form-control")
+          .simulate("change", { target: { value: courseRun["id"] } })
+        inner.update()
+        assert.isFalse(
+          inner
+            .find("button.enroll-now-free")
+            .at(0)
+            .prop("disabled")
+        )
+        assert.isFalse(
+          inner
+            .find("button.btn-upgrade")
+            .at(0)
+            .prop("disabled")
+        )
       } else {
         assert.isFalse(selectorControl.exists())
+        assert.isFalse(
+          inner
+            .find("button.enroll-now-free")
+            .at(0)
+            .prop("disabled")
+        )
+        assert.isFalse(
+          upgradeForm
+            .find("button.btn-upgrade")
+            .at(0)
+            .prop("disabled")
+        )
       }
     })
   })
+
+  it(`renders enrollment dialog for a course with multiple non-upgradable active runs`, async () => {
```

Comment:
Can you update the description of this test, and the one above that you modified, to indicate what the test is ensuring happens?